xamarin/Xamarin.Forms
# UIStatusBarStyle now changes correctly when a device is in dark mode

### Description of Change ###
With the introduction of iOS 13, dark mode and the new UIStatusBarStyle.DarkContent enum status bar color set automatically from a NavigationPage has been incorrect. When the device is in dark mode and tries to update the StatusBarStyle to UIStatusBarStyle.Default this property behaves differently depending on if the device is in dark/light mode (or if the app is forced into light/dark mode with the UIUserInterfaceStyle key in Info.plist) and we get an unexpected status bar style.

Instead now when running on iOS13 it will use the new UIStatusBarStyle.DarkContent enum instead of UIStatusBarStyle.Default to display the expected UIStatusBarStyle.

### Issues Resolved ### 
- fixes #8294

### API Changes ### 
None

### Platforms Affected ### 
- iOS

### Behavioral/Visual Changes ###
On iOS 13 when a device is using dark mode the correct UIStatusBarStyle is now set. 

### Before/After Screenshots ### 

![beforeafter_small](https://user-images.githubusercontent.com/904737/70368510-8b3d5100-18ff-11ea-98c6-946918fd4100.png)


### Testing Procedure ###
Issue8294 was added to Xamarin.Forms.Controls.Issues.
To test, change UIViewControllerBasedStatusBarAppearance to false in the Xamarin.Forms.ControlGallery.iOS Info.plist. This allows us to not rely on each ViewController to specify its UIStatusBarStyle and instead we can set it with `UIApplication.SharedApplication.StatusBarStyle = ...;`

On an iOS device not in dark mode launch the control gallery and open issue 8294. 
Tapping light and dark you should see the status bar text color change correctly.
Now close the control gallery app, put the device in dark mode, launch the control gallery app and repeat the above steps. 

### PR Checklist ###
